tangos/scripts/manager.py

```diff
@@ -207,9 +210,21 @@ def remove_duplicates(options):
         )
     """)).rowcount
 
+    count_links = session.execute(dedent("""
+        DELETE FROM halolink
+        WHERE id NOT IN (
+            SELECT * FROM (
+                SELECT MAX(id)
+                FROM halolink
+                GROUP BY halo_from_id, halo_to_id, weight
```

Comment:
I think this should also be grouped by the name of the link (`relation_id`). Although not common, it's logically possible some people would have two links which are basically the same but named something different, because they represent a different relationship.

I also wonder what the logic for grouping by weight is. In principle I think it makes no sense to have two links between the same halos, with the same name, but different weights. Ideally a cull would keep the one with the highest weight. This is surprisingly difficult but you will find it implemented in `tangos.util.sql_argmax.delete_non_maximal_rows`. Willing to be convinced this isn't necessary, but from what I can see it might be the neatest thing to do.

Reply:
I agree that maximal duplicates should also be grouped by `relation_id`, I didn't think of it, thanks.

As for grouping by `weight`, my idea was that the safest approach is the "maximal" approach. 

I agree with you that conceptually, linking the same halos with the same name but different weight makes no sense. However, it's unclear to me which link should be kept between the most recent addition to the database (as seems to be done for `halo properties`) and the highest weight link (which might be old/wrong and overridden by more recent calculations). 

The current implementation currently leaves deletion of non-maximal duplicates as a user's choice, but we could easily copy the behaviour of `halo properties` and simply keep the most recent update.

Let me know what you think.